functions-and-operators/encryption-and-compression-functions.md

```diff
@@ -10,32 +10,420 @@ TiDB supports most of the [encryption and compression functions](https://dev.mys
+
+```sql
+SELECT AES_ENCRYPT(0x616263,'secret');
+```
+
+```
++----------------------------------------------------------------+
+| AES_ENCRYPT(0x616263,'secret')                                 |
++----------------------------------------------------------------+
+| 0x28409970815CD536428876175F1A4923                             |
++----------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+### [`COMPRESS()`](https://dev.mysql.com/doc/refman/8.0/en/encryption-functions.html#function_compress)
+
+The `COMPRESS(expr)` function returns a compressed version of the input data `expr`.
+
+- If the argument is `NULL`, the function returns `NULL`.
+- If the argument is an empty string, the function returns a zero-length value.
+
+For non-zero length argument, the function returns a binary string with the following structure:
+
+- Bytes 0 to 3: the uncompressed length
+- Bytes 4 to the end: the zlib compressed data
+
+```sql
+SELECT COMPRESS(0x414243);
+```
+
+```
++------------------------------------------+
+| COMPRESS(0x414243)                       |
++------------------------------------------+
+| 0x03000000789C72747206040000FFFF018D00C7 |
++------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+In this output, `0x03000000` represents the uncompressed length (3) and `0x789C72747206040000FFFF018D00C7` is the zlib compressed data.
+
+An example of using Python to decode this outside of TiDB:
+
+```python
+import codecs
+import zlib
+
+data = codecs.decode('03000000789C72747206040000FFFF018D00C7','hex')
+print(int.from_bytes(data[:4], byteorder='little'))  # 3
+print(zlib.decompress(data[4:]))  # b'ABC'
+```
+
+For short strings, `COMPRESS()` might return more bytes than the input. The following example shows that a string of 100 `a` characters compresses to 19 bytes.
+
+```sql
+WITH x AS (SELECT REPEAT('a',100) 'a')
+SELECT LENGTH(a),LENGTH(COMPRESS(a)) FROM x;
+```
+
+```
++-----------+---------------------+
+| LENGTH(a) | LENGTH(COMPRESS(a)) |
++-----------+---------------------+
+|       100 |                  19 |
++-----------+---------------------+
+1 row in set (0.00 sec)
+```
+
+### [`MD5()`](https://dev.mysql.com/doc/refman/8.0/en/encryption-functions.html#function_md5)
+
+The `MD5(expr)` function calculates a 128-bit [MD5](https://en.wikipedia.org/wiki/MD5) hash for the given argument `expr`.
+
+```sql
+SELECT MD5('abc');
+```
+
+```
++----------------------------------+
+| MD5('abc')                       |
++----------------------------------+
+| 900150983cd24fb0d6963f7d28e17f72 |
++----------------------------------+
+1 row in set (0.00 sec)
+```
+
+### [`PASSWORD()`](https://dev.mysql.com/doc/refman/5.7/en/encryption-functions.html#function_password)
+
+> **Warning:**
+>
+> This function is deprecated in MySQL 5.7 and removed in MySQL 8.0. It is deprecated in TiDB. It is not recommended to use this function.
```

Comment:
```suggestion
> This function is deprecated in MySQL 5.7 and removed in MySQL 8.0. It is deprecated in TiDB vx.y.0. It is not recommended to use this function.
```
We need to include this deprecation in the release notes.

Reply:
It seems that PASSWORD() has been deprecated since 2018 https://github.com/pingcap/tidb/pull/6000